### PR TITLE
feat(lrcnormalize): LRC multi-timestamp expander + tag classifier (#469)

### LINKED ISSUE #469: LRC normalizer: expand compressed multi-timestamp lines (foundation)
[mode: plan]
[model: sonnet]
[effort: medium]

## Summary

Add a shared, pure LRC normalizer that expands compressed multi-timestamp lines into one cue per timestamp and classifies ID-tag lines vs. real cues.

## Problem / Motivation

The LRC format allows one line to carry several leading timestamps for a repeated lyric, e.g. `[t1][t2][t3]Chorus line`. canticle's LRC-text parsing keeps only the first timestamp and leaves the rest embedded in the line text, so the compressed form is both mis-parsed and re-emitted. A single, well-tested transform is needed as the foundation for two consumers: the write path (emit clean, universally-readable LRC) and the future upgrade path (flatten a candidate LRC to comparable text for match scoring).

## Proposed Solution

New internal package (e.g. `internal/lrcnormalize`):
- Parse an `.lrc` body (or a `models.Synced`) into an expanded cue model.
- Split N stacked leading timestamps on a line into N cues with identical text.
- Classify `[key:value]` ID-tag lines (`[ar:]`, `[ti:]`, `[ve:]`, `[length:]`, ...) distinctly from real cues.
- Pure (no I/O). Table + fuzz tests.

## Acceptance Criteria

- [ ] N stacked timestamps on a line expand into N cues with identical text
- [ ] ID-tag lines are classified/handled distinctly from cues
- [ ] Pure transform (no I/O); table + fuzz tests
- [ ] Usable by the write path and the upgrade path

## Additional Context

Foundation issue; blocks the LRC-normalize write/backfill work and is reused by the upgrade middleware. Design notes were captured out-of-repo for a future session.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- New pure package `internal/lrcnormalize` (foundation for #470 / #472): `ParseBody` expands consecutive leading LRC timestamps (`[t1][t2]text`) into one cue per timestamp with shared, trimmed text, sorts cues ascending (stable), and classifies `[key:value]` header tags out of the cue stream.
- `Expand` re-splits an already-parsed `models.Synced` whose cue text still carries an embedded leading stamp (the single-timestamp-regex parse bug), splitting only text-embedded stamps and preserving each cue's original `Time` verbatim (no lossy re-render); idempotent on clean input.
- No consumer wired (issue scope is "usable by", not "used by"). Cue-text trimming and the timestamp tolerance/`normalizeHundredths` match `internal/petitlyrics`, so the package is a drop-in for that lane.

Design decisions (adjacent + whitespace-separated expansion, drop-orphan lines) are grounded in a host-side GNU-grep census of the 35,168-file production library: 74,316 stacked lines (32% of files), 0 synced files with stray non-cue lines.

## Linked issue

Closes #469

## Gates (run locally; CI enforces)

- [x] `make gate` — conflict-markers, gofmt, templ+tailwind generate, build, `go test -race`, patch coverage, coverage-floor, codecov dry-run, golangci-lint, actionlint, govulncheck
- [x] Package coverage 93.8%; patch coverage 91.00% (threshold 70%)
- [x] golangci-lint 0 issues; govulncheck 0 reachable
- [x] Adversarial pre-push review (2 Important findings fixed: cue-text trim divergence from petitlyrics, and a lossy `Expand` re-render that dropped `Total`-only / minute≥100 cues; both now test-covered)

## Test plan

- [x] Table tests: single/stacked/out-of-order/whitespace-separated stamps, tag classification, ms precision, `♪` marker, orphan+blank drop, single-digit-seconds drop, empty/tags-only bodies
- [x] `Expand` tests: embedded-stamp split, idempotency, `Total`-only preservation, minute≥100 outer-cue preservation
- [x] `FuzzParseBody` (30s clean): asserts sorted, self-consistent timestamps and that no expanded cue text retains a leading stamp

## Summary by CodeRabbit

- **New Features**
  - Added improved LRC normalization that expands lines with multiple timestamps into separate synchronized cues.
  - Recognizes metadata tags separately from lyric cues.
  - Handles embedded timestamps, cue ordering, trimming, and fractional-second normalization.

- **Tests**
  - Added comprehensive unit and fuzz testing for timestamp parsing, metadata handling, ordering, malformed input, and edge cases.

- **Documentation**
  - Documented the new LRC normalization capability in the package catalog.

### CODE REVIEW

**Comment by sydlexius** [comment]
@coderabbitai review

File: internal/lrcnormalize/lrcnormalize.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
+// Package lrcnormalize expands compressed multi-timestamp LRC lines into one
+// cue per timestamp and classifies [key:value] ID-tag lines distinctly from
+// real cues. It is a pure transform (no I/O) shared by the LRC-text provider
+// parse lanes, the write/backfill path, and the upgrade match-scorer.
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/lrcnormalize/lrcnormalize.go]: Fixed in 469f68d. Reworded the package doc (lrcnormalize.go) and the CLAUDE.md catalog line to describe the parse / write-backfill / scoring paths as the intended foundation, with an explicit "no consumer wired yet" note.

**Comment by sydlexius** [comment]
Fixed in 469f68d. The single actionable finding (docs describing the parse/write-backfill/scoring paths as current integrations rather than the intended, not-yet-wired foundation) is addressed in both flagged locations: the package doc comment and CLAUDE.md.

_(Addressing @coderabbitai's review 4710442437.)_

**Comment by sydlexius** [comment]
@coderabbitai resolve
